src/AudacityApp.cpp

```diff
@@ -238,12 +238,12 @@ DEFINE_EVENT_TYPE(EVT_LANGUAGE_CHANGE);
 
 #if 0
 #ifdef __WXGTK__
-static void wxOnAssert(const wxChar *fileName, int lineNumber, const wxChar *msg)
+static void wxOnAssert(const wxString &fileName, int lineNumber, const wxString &msg)
 {
-   if (msg)
-      printf("ASSERTION FAILED: %s\n%s: %d\n", (const char *)wxString(msg).mb_str(), (const char *)wxString(fileName).mb_str(), lineNumber);
+   if (!msg.empty())
+      wxPrintf("ASSERTION FAILED: %s\n%s: %d\n", msg, fileName, lineNumber);
    else
-      printf("ASSERTION FAILED!\n%s: %d\n", (const char *)wxString(fileName).mb_str(), lineNumber);
+      wxPrintf("ASSERTION FAILED!\n%s: %d\n", fileName, lineNumber);
 
```

Comment:
This is an example of the change described here: http://docs.wxwidgets.org/trunk/classwx_string.html#string_vararg which basically boils down to the "right" solution being different to what worked in wx2.8.